source/robot_model/test/tests/test_kinematics.cpp

```diff
@@ -148,3 +148,36 @@ TEST_F(RobotModelKinematicsTest, TestInverseKinematics) {
     EXPECT_TRUE(joint_twist.get_velocities().isApprox(test_configs[config].get_velocities()));
   }
 }
+
+TEST_F(RobotModelKinematicsTest, TestInRange) {
+  
+  state_representation::JointPositions joint_positions("robot", 7);
+  state_representation::JointVelocities joint_velocities("robot", 7);
+  state_representation::JointTorques joint_torques("robot", 7);
+  state_representation::JointState joint_state("robot", 7);
+  
+  joint_positions.set_positions({2.648782, -0.553976, 0.801067, -2.042097, -1.642935, 2.946476, 1.292717});
+  joint_velocities.set_velocities({-0.059943, 1.667088, 1.439900, -1.367141, -1.164922, 0.948034, 2.239983});
+  joint_torques.set_torques({-0.329909, -0.235174, -1.881858, -2.491807, 0.674615, 0.996670, 0.345810});
+
+  joint_state.set_positions({2.648782, -0.553976, 0.801067, -2.042097, -1.642935, 2.946476, 1.292717});
+  joint_state.set_velocities({-0.059943, 1.667088, 1.439900, -1.367141, -1.164922, 0.948034, 2.239983});
+  joint_state.set_torques({-0.329909, -0.235174, -1.881858, -2.491807, 0.674615, 0.996670, 0.345810});
+
+  EXPECT_TRUE(franka->in_range(joint_positions));
+  EXPECT_TRUE(franka->in_range(joint_velocities));
+  EXPECT_TRUE(franka->in_range(joint_torques));
+  EXPECT_TRUE(franka->in_range(joint_state));
+
+  joint_positions.set_positions({-0.059943, 1.667088, 1.439900, -1.367141, -1.164922, 0.948034, 1000});
```

Comment:
It's hard to tell from the numbers, but I imagine these values cover both the positive and negative sides of the joint limits? Just to make sure it's checking both cases.

Reply:
The one that exceeds the limits are one at the time, positive or negative, to check both of the possibilities